boards/xtensa/esp32/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+
+if(CONFIG_BOOTLOADER_ESP_IDF)
+  include(ExternalProject)
+
+  ## we use hell-world project, but I think any can be used.
```

Comment:
Just a note: It should be fine to use the partition table for the hello-world project; the way I see this, `CONFIG_BOOTLOADER_ESP_IDF` is more of a stepping stone to improve hardware support in ESP32, so this can be improved at a later date to allow custom partition tables, or the use of the more mcuboot bootloader (it has less features than the Espressif bootloader, but is more consistent with the Zephyr ecosystem.)